Dynamic Programming/Ugly Numbers

```diff
@@ -0,0 +1,47 @@
+// C++ program to find n'th Ugly number 
+# include<bits/stdc++.h> 
```

Comment:
please don't use **bits/stdc++.h** because this is quite Linux specific and slows down the compiler process. _(Specified in [README](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/README.md))_
and add **.cpp** file extension too.

Same changes requested for PR #109, #110, #111 & #112